File: ios/App/Sources/Core/Domain/NavCaddy/PressureContext.swift
import Foundation

/// Represents the pressure context during a shot.
///
/// Spec R5, Q4: Pressure detection combines user toggle + scoring inference.
struct PressureContext: Codable, Hashable {
    /// User manually tagged this shot as high pressure
    let isUserTagged: Bool

    /// System inferred pressure from scoring context (e.g., close match, tournament)
    let isInferred: Bool

    /// Optional description of the scoring context
    /// Examples: "leading by 1", "tournament mode", "trailing by 2"
    let scoringContext: String?

    init(
        isUserTagged: Bool = false,
        isInferred: Bool = false,
        scoringContext: String? = nil
    ) {
        self.isUserTagged = isUserTagged
        self.isInferred = isInferred
        self.scoringContext = scoringContext
    }

    /// Returns true if there is any pressure context (user or inferred)
    var hasPressure: Bool {
        isUserTagged || isInferred
    }
}

File: android/app/src/main/java/com/example/app/domain/navcaddy/models/PressureContext.kt
package caddypro.domain.navcaddy.models

import com.google.gson.annotations.SerializedName

/**
 * Domain model representing pressure context for a shot.
 *
 * Pressure context helps identify patterns that emerge under specific
 * competitive or scoring situations.
 *
 * Spec reference: navcaddy-engine.md R5, Q4
 *
 * @property isUserTagged Whether the user explicitly tagged this as a pressure shot
 * @property isInferred Whether pressure was inferred from scoring context
 * @property scoringContext Description of scoring situation (e.g., "leading by 1", "tournament mode")
 */
data class PressureContext(
    @SerializedName("is_user_tagged")
    val isUserTagged: Boolean = false,

    @SerializedName("is_inferred")
    val isInferred: Boolean = false,

    @SerializedName("scoring_context")
    val scoringContext: String? = null
) {
    /**
     * Indicates whether any pressure condition is present.
     */
    val hasPressure: Boolean
        get() = isUserTagged || isInferred
}
